### PR TITLE
Introduce a netboot class

### PR DESCRIPTION
**Pull request by ekohl**
Relates to https://github.com/theforeman/puppet-foreman_proxy/issues/346

Currently I've added a separate EFI params class but I'm not sure this is the best design. We might want to pull the EFI install type to a top level.

### CODE REVIEW

**Comment by ekohl** [comment]
@mmoll I'll fix the tests but I would like some feedback on the general design. Mostly about introducing the ```foreman_proxy::efi::params``` class. Should that be in the main params? And do we want to expose the option?

My preference is that foreman-bootloaders is added to the supported releases (1.14 and 1.15 right now) and we switch over the RH option from grub to package. It should be fair to assume it works and most users won't need to.

File: manifests/tftp/efi.pp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# EFI support for TFTP
+class foreman_proxy::tftp::efi (
```
Comment (lzap) [inline manifests/tftp/efi.pp]: The class name is little bit confusing, our initial goal was to support EFI but we also deploy ELF binaries for BOOTP (PPC) and BIOS firmware here. Better candidate would be perhaps "netboot".

File: manifests/tftp/efi.pp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        mode   => '0644',
+        source => '/boot/efi/EFI/redhat/grub.efi',
+      }
+
+      file {"${root}/grub/shim.efi":
+        ensure => 'link',
+        target => 'grubx64.efi',
+      }
+    }
+    'debian_grub': {
```
Comment (lzap) [inline manifests/tftp/efi.pp]: Although we do not provide debian package yet, tarballs are downloadable from http://downloads.theforeman.org/foreman-bootloaders/ for debian as well. Might be also an option for debian users.

File: manifests/tftp/efi.pp (hunk 1 above)
Comment (ekohl) [reply manifests/tftp/efi.pp]: Ok, that makes sense.

File: manifests/tftp/efi.pp (hunk 2 above)
Comment (ekohl) [reply manifests/tftp/efi.pp]: I do wonder why there's the OS name in the package name. Why can't it just be yum/apt install foreman-bootloaders?

**Comment by mmoll** [comment]
From a quick glance, I'd also say "netboot" is a good name, and if it isn't really required, there should be as little exposed subclass parameters as possible.

**Comment by ekohl** [comment]
Renamed efi to netboot.

**Comment by ekohl** [comment]
I've refactored it a bit. My goal is that when it's packaged we can change params to set ```packages => ['foreman-bootloaders-redhat'], grub_installation_type => 'none'```.

**Comment by ekohl** [comment]
The tests are now green and ready for a final review.

**Comment by mmoll** [comment]
please rebase ;)

**Comment by ekohl** [comment]
There we go :)

**Comment by ekohl** [comment]
This is ready for a review

**Comment by mmoll** [comment]
merged, bedankt @ekohl!

File: manifests/tftp/efi.pp (hunk 2 above)
Comment (lzap) [reply manifests/tftp/efi.pp]: Distributions carries a lot (hundreds) of patches in grub2, redhat distribution is very different from Debian/Ubuntu. These builds were taken from redhat, thus the name. You can install debian versions next to those when we make them available.

**Comment by lzap** [comment]
@ekohl ++ thanks for doing this, I filed another ticket the other day, it would be good to create the grub the same way as debian branch does via the make image command instead of simple copy.

**Comment by ekohl** [comment]
> @ekohl ++ thanks for doing this, I filed another ticket the other day, it would be good to create the grub the same way as debian branch does via the make image command instead of simple copy.

That's on my (rather long) todo list

**Pull request merged**